src/sinks/splunk_hec.rs

```diff
@@ -122,6 +123,19 @@ impl HttpSink for HecSinkConfig {
     fn encode_event(&self, mut event: Event) -> Option<Self::Input> {
         self.encoding.apply_rules(&mut event);
 
+        let sourcetype = self
+            .sourcetype
+            .as_ref()
+            .map(|sourcetype| {
+                sourcetype
+                    .render_string(&event)
+                    .map_err(|missing_keys| {
+                        emit!(SplunkSourceTypeMissingKeys { keys: missing_keys });
+                    })
+                    .ok()
+            })
+            .unwrap_or(None);
```

Comment:
To avoid nesting `Option`s here you should be able to replace `map` with `and_then` and get rid of the unwrap.

Reply:
And so I can :smile: That is better. 3271c029